nix/lib/purescript.nix

```diff
@@ -1,129 +1,52 @@
 { stdenv
-, lib
-, linkFarm
-, cacert
 , nodejs
-, nodePackages
-, python2
-, git
-, fetchurl
-, yarn2nix-moretea
-, nodejs-headers
 , easyPS
+, nix-gitignore
+, buildNodeModules
 }:
 
-{ psSrc
+{
+  # path to generated purescript sources
+  psSrc
+  # path to project sources
 , src
+  # name of the project
 , name
-, yarnLock ? src + "/yarn.lock"
-, yarnNix ? src + "/yarn.nix"
-, packageJSON ? src + "/package.json"
-, additionalPurescriptSources ? [ ]
+  # additional dependencies required to build node_modules (e.g available during `npm install`)
+, additionalNpmBuildInputs ? [ ]
+  # packages as generated by psc-pacakge2nix
 , packages
+  # spago packages as generated by spago2nix
 , spagoPackages
+  # web-common project
 , webCommon
-, checkPhase ? "yarn --offline test"
-, passthru ? { }
+  # test script
+, checkPhase ? ""
 }:
 let
-  # Note that these binaries are both versioned with the node-sass version *and* the NODE_MODULE_VERSION
-  # version, so you may need to change that when updating to a new version of nodejs.
-  # For reasons beyond my ken, these do *not* work if you try and get them with niv. Mysteriously you get snytax
-  # errors in the binding, even though diffing the files they look the same. I gave up and left them like ths.
-  nodeSassBinLinux = fetchurl {
-    url = "https://github.com/sass/node-sass/releases/download/v5.0.0/linux-x64-72_binding.node";
-    sha256 = "0ja9b462dzxmkqm15zc0flv33cq16jy86vigj7gbmmad2pgkryq8";
-  };
-  nodeSassBinDarwin = fetchurl {
-    url = "https://github.com/sass/node-sass/releases/download/v5.0.0/darwin-x64-72_binding.node";
-    sha256 = "0b079cygjr3i70hwvb1ik83z2lxbbl1daqky0i8mnbibxbgr4jb7";
-  };
-  nodeSassBinDir = linkFarm "node-sass-bin-dir" [
-    { name = "linux-x64-72/binding.node"; path = nodeSassBinLinux; }
-    { name = "darwin-x64-72/binding.node"; path = nodeSassBinDarwin; }
-  ];
-
-  packagesJson = "${src}/packages.json";
-
-  cleanSrcs = lib.cleanSourceWith {
-    filter = lib.cleanSourceFilter;
-    src = lib.cleanSourceWith {
-      filter = (path: type: !(lib.elem (baseNameOf path)
-        [ ".spago" ".spago2nix" "generated" "generated-docs" "output" "dist" "node_modules" ".psci_modules" ".vscode" ]));
-      inherit src;
-    };
-  };
-
-  purescriptSources = [
-    "src/**/*.purs"
-    "test/**/*.purs"
-    "generated/**/*.purs"
-    ".spago/*/*/src/**/*.purs"
-  ] ++ additionalPurescriptSources;
+  # Cleans the source based on the patterns in ./.gitignore and the additionalIgnores
+  cleanSrcs = nix-gitignore.gitignoreSource [ "/*.adoc" "/*.nix" ] src;
```

Comment:
All the clients have local `.gitignore` files so this works nicely. The only problem would be if this was to be used elsewhere and a gitignore was missing.

Reply:
I spent a while trying the various nix-gitignore things, I don't remember why we're using the one we're using, unfortunately. I vaguely recall having some issues with the hercules-ci one...

Reply:
Alright, i wasn't aware. The one we are currently suggests using the hercules one instead but it *works* so .. no pressure to switch for now.